client/packageutils.c

```diff
@@ -933,6 +944,87 @@ TDNFCheckProtectedPkgs(
     return dwError;
 }
 
+uint32_t
+TDNFGetAvailableCacheBytes(
+    PTDNF_CONF pConf,
+    uint64_t* pqwAvailCacheDirBytes
+    )
+{
+    uint32_t dwError = 0;
+    struct statfs tmpStatfsBuffer = {0};
+
+    if(!pConf || !pConf->pszCacheDir || !pqwAvailCacheDirBytes)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    if (statfs(pConf->pszCacheDir, &tmpStatfsBuffer) != 0)
+    {
+        dwError = errno;
+        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+    }
+
+    *pqwAvailCacheDirBytes = tmpStatfsBuffer.f_bsize * tmpStatfsBuffer.f_bavail;
+
+cleanup:
+    return dwError;
+
+error:
+    if(pqwAvailCacheDirBytes)
+    {
+        *pqwAvailCacheDirBytes = 0;
+    }
+    goto cleanup;
+}
+
+uint32_t
+TDNFCheckDownloadCacheBytes(
+    PTDNF_SOLVED_PKG_INFO pSolvedPkgInfo,
+    uint64_t qwAvailCacheBytes
+    )
+{
+    uint32_t dwError = 0;
+    uint64_t qwTotalDownloadSizeBytes = 0;
+    uint8_t byPkgIndex = 0;
+    PTDNF_PKG_INFO pPkgInfo = NULL;
+
+    if(!pSolvedPkgInfo)
```

Comment:
can we club?
if(!pSolvedPkgInfo || !pSolvedPkgInfo->nNeedDownload)

Reply:
I agree and would like to put both condition in one `if` statement, but one of those cases, namely `if(!pSolvedPkgInfo)`, is an error and should return a non-zero error code. But the second condtion, `if(!pSolvedPkgInfo->nNeedDownload)` means that there are no packages to download and so we can simply return without error. That's why I placed them separately.

Reply:
Addressed, Thanks!